File: JavaScript_Basic_04_2023_SolutionsOfProblems/05.Ex_PB-JS-While-Loop/P06.Cake.js
function cake(input){
    let cakePieces=Number(input[0])*Number(input[1]);
    let i=2;
    while(input[i]!=="STOP"&&cakePieces>0){
        cakePieces-=Number(input[i]);
        i++;
    }
    if(cakePieces>0){
        console.log(`${cakePieces} pieces are left.`);
    }else{
        console.log(`No more cake left! You need ${Math.abs(cakePieces)} pieces more.`);
    }
}
cake(["10",
"2",
"2",
"4",
"6",
"STOP"])

File: SoftUni_JAVA_BASIC_Jan_2023_SolutionsOfProblems/P05Ex_While_Loop/P06_Cake.java
package P05Ex_While_Loop;

import java.util.Scanner;

public class P06_Cake {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        int widthCake = Integer.parseInt(scanner.nextLine());
        int lengthCake = Integer.parseInt(scanner.nextLine());
        int piecesCake = widthCake * lengthCake;
        int getPieces = 0;
        String input = "";
        while (piecesCake > 0) {
            input = scanner.nextLine();
            if (input.equals("STOP")) {
                break;
            }
            getPieces = Integer.parseInt(input);
            piecesCake -= getPieces;

        }
        if (input.equals("STOP")) {
            System.out.printf("%d pieces are left.", piecesCake);
        } else {
            System.out.printf("No more cake left! You need %d pieces more.", -piecesCake);
        }
    }
}
